src/api/Workspace.ts

```diff
@@ -1,24 +1,61 @@
 import { StartRequest } from './methods/start';
-import { ServiceRequest, ServiceResponse } from './methods/service';
-import { RegisterRequest } from './methods/register';
+import { ServicesMap, ServicesRequest } from './methods/services';
+import { RegisterServiceRequest } from './methods/registerService';
+import { ComponentsMap, ComponentsRequest } from './methods/components';
+import { ConfigRequest } from './methods/config';
 
-interface Workspace {
-  // Load all the services from url
-  // Load the layout and render components inside
+/**
+ * Workspace is the core service of CapsulaHub, it is responsible for :
+ * - Loading services and components that are included in its configuration (START)
+ * - Allowing them to register themselves (REGISTER)
+ * - Letting them communicate together (SERVICES/COMPONENTS)
+ * - Getting their own configuration (CONFIG)
+ */
+export interface Workspace {
+  /**
+   * Loading all services and components included in workspace configuration
+   * Reject in case :
+   * - Workspace already started
+   * - Workspace start already in progress
+   * - An error with importing a service occurred
+   * - An error with importing a component occurred
+   * - An error with registering a component occurred
+   * @param startRequest
+   */
   start(startRequest: StartRequest): Promise<void>;
 
-  // Returns proxy of a registered service
-  // Can be used by any service during run time to get a proxy of another service
-  service(serviceRequest: ServiceRequest): Promise<ServiceResponse>;
+  /**
+   * Getting promises to each service that has been loaded in the workspace
```

Comment:
We actually get a promise, that includes the map of services with the promise of each service included